File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/interop/service/PingInteropSpec.kt
package stasis.test.client_android.lib.interop.service

import stasis.client_android.lib.model.server.api.responses.Ping
import stasis.test.client_android.lib.interop.InteropSpec
import java.util.UUID

class PingInteropSpec : InteropSpec({
    "Ping interop" should {
        "decode and re-encode a ping" {
            assert(
                domain = "service",
                resource = "Ping",
                matches = Ping(
                    id = UUID.fromString("a55bf2c4-c270-4d8b-806b-9f993e8d415a")
                )
            )
        }
    }
})

File: shared/src/test/scala/stasis/test/specs/unit/shared/interop/service/PingInteropSpec.scala
package stasis.test.specs.unit.shared.interop.service

import java.util.UUID

import stasis.shared.api.Formats._
import stasis.shared.api.responses.Ping
import stasis.test.specs.unit.shared.interop.InteropSpec

class PingInteropSpec extends InteropSpec {
  "Ping interop" should "decode and re-encode a ping" in {
    assert(
      domain = "service",
      resource = "Ping",
      matches = Ping(
        id = UUID.fromString("a55bf2c4-c270-4d8b-806b-9f993e8d415a")
      )
    )
  }
}
